gcloud-java-contrib/gcloud-java-nio/src/test/java/com/google/gcloud/storage/contrib/nio/ITGcsNio.java

```diff
@@ -0,0 +1,325 @@
+import java.nio.ByteBuffer;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.file.CopyOption;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.List;
+import java.util.Random;
+
+
+/**
+ * Integration test for gcloud-nio. This test actually talks to GCS (you need an account).
+ * Tests both reading and writing.
+ */
+@RunWith(JUnit4.class)
+public class ITGcsNio {
+
+  //
+  // The variables below are filled in via environment variables
+  // (of the same name).
+  // You *must* set these environment variables before running this test or it will fail.
+  // The test also relies on you using the gcloud command-line utility to log into
+  // an account that has read access to the bucket you'll point the tests to.
+  //
+  // The instructions for how to get the Service Account JSON Key are
+  // at https://cloud.google.com/storage/docs/authentication?hl=en#service_accounts
+  //
+  // The short version is this: go to cloud.google.com/console,
+  // select your project, search for "API manager", click "Credentials",
+  // click "create credentials/service account key", new service account,
+  // JSON. The contents of the file that's sent to your browsers is your
+  // "Service Account JSON Key".
+  //
+
+  // The bucket we'll read the files from (e.g. "abucket")
+  private final String TEST_NIO_BUCKET;
+  // A first file we can read (can be small) (e.g. "folder/file.txt")
+  private final String TEST_NIO_SML;
+  // The size of the small file, in bytes
+  private final int TEST_NIO_SML_SIZE;
+  // A second file we can read (can be large)
+  private final String TEST_NIO_LGE;
+  // A non-existant file we can write to. We're going to add a suffix to it
+  // just in case multiple tests are running concurrently.
+  private final String TEST_NIO_WRITE_PREFIX;
+  // The project that the bucket belongs to.
+  // This ought to match the project_id entry in the service account json key below.
+  private final String TEST_NIO_PROJECT;
+  // the full text of the service account key.
+  // e.g. {"type":"service_account","project_id":...
+  // Naturally you don't want to share this since it gives people access to this project.
+  private final String TEST_NIO_SERVICE_ACCOUNT_JSON_KEY;
+
+  private final Random rnd;
+
+  private static final List<String> FILE_CONTENTS = ImmutableList.of(
+      "Tous les êtres humains naissent libres et égaux en dignité et en droits.",
+      "Ils sont doués de raison et de conscience et doivent agir ",
+      "les uns envers les autres dans un esprit de fraternité.");
+
+
+
+  /**
+   * Sets up the test according to the provided env. vars.
+   */
+  public ITGcsNio() {
+    TEST_NIO_BUCKET = getEnv("TEST_NIO_BUCKET");
```

Comment:
I think you should use RemoteGcsHelper to create a bucket for the test (and delete at the end). see ITStorageTest and
[this](https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/TESTING.md#testing-code-that-uses-storage).

Also, we should follow the same procedure for project and account credentials.
We should create the necessary files as part of the test setup rather then depending on them to exists.